crates/tower-cmd/src/run.rs

```diff
@@ -374,114 +474,149 @@ fn get_app_name(cmd: Option<(&str, &ArgMatches)>) -> Option<String> {
-            }
+    let res = api::export_secrets(&config, env, false, public_key).await.map_err(|err| {
+        debug!("API error fetching secrets: {:?}", err);
+        Error::FetchingSecretsFailed
+    })?;
+    let mut secrets = HashMap::new();
 
-            Ok(secrets)
-        },
-        Err(err) => {
-            output::tower_error(err);
-            Err(Error::FetchingSecretsFailed)
-        }
+    for secret in res.secrets {
+        // we will decrypt each property and inject it into the vals map.
+        let decrypted_value = crypto::decrypt(private_key.clone(), secret.encrypted_value.to_string())?;
+        secrets.insert(secret.name, decrypted_value);
     }
+
+    Ok(secrets)
 }
 
 /// get_catalogs manages the process of exporting catalogs, decrypting their properties, and
 /// preparting them for injection into the environment during app execution
 async fn get_catalogs(config: &Config, env: &str) -> Result<HashMap<String, String>, Error> {
     let (private_key, public_key) = crypto::generate_key_pair();
 
-    match api::export_catalogs(&config, env, false, public_key).await {
-        Ok(res) => {
-            let mut vals = HashMap::new();
-
-            for catalog in res.catalogs {
-                // we will decrypt each property and inject it into the vals map.
-                for property in catalog.properties {
-                    let decrypted_value = crypto::decrypt(private_key.clone(), property.encrypted_value.to_string())?;
-                    let name = create_pyiceberg_catalog_property_name(&catalog.name, &property.name);
-                    vals.insert(name, decrypted_value);
-                }
-            }
-
-            Ok(vals)
-        },
-        Err(err) => {
-            output::tower_error(err);
-            Err(Error::FetchingCatalogsFailed)
+    let res = api::export_catalogs(&config, env, false, public_key).await.map_err(|err| {
+        debug!("API error fetching catalogs: {:?}", err);
+        Error::FetchingCatalogsFailed
+    })?;
+    let mut vals = HashMap::new();
+
+    for catalog in res.catalogs {
+        // we will decrypt each property and inject it into the vals map.
+        for property in catalog.properties {
+            let decrypted_value = crypto::decrypt(private_key.clone(), property.encrypted_value.to_string())?;
+            let name = create_pyiceberg_catalog_property_name(&catalog.name, &property.name);
+            vals.insert(name, decrypted_value);
         }
     }
+
+    Ok(vals)
 }
 
 /// load_towerfile manages the process of loading a Towerfile from a given path in an interactive
 /// way. That means it will not return if the Towerfile can't be loaded and instead will publish an
 /// error.
-fn load_towerfile(path: &PathBuf) -> Towerfile {
-    Towerfile::from_path(path.clone()).unwrap_or_else(|err| {
+fn load_towerfile(path: &PathBuf) -> Result<Towerfile> {
+    Towerfile::from_path(path.clone()).map_err(|err| {
```

Comment:
Before, `unwrap_or_else` tries to take the result of a function that returns Result<Self, Error>, and replaces the error result with this new value... which in this case just called `std::process::exit(1)` (so I guess didn't replace it?). Now we want to keep the Result type, so we want to specifically call a function when we get an error response so that we can add some debug info, which is what `map_err` does in rust (i.e. it "maps" a function to an "err" value... personally I would have named this function something else but it's a pretty well known rust idiom by now).